packages/list/index.js

```diff
@@ -82,219 +85,188 @@ export default class List extends Component {
     this.foundation_.destroy();
   }
 
-  initListItem = (id) => {
-    const {listItemAttributes, listItemChildrenTabIndex, listItemClassList} = this.state;
-    listItemAttributes[id] = {
-      'aria-selected': false,
-      'tabIndex': -1,
-    };
-    listItemChildrenTabIndex[id] = -1;
-    listItemClassList[id] = new Set();
-    this.setState({listItemAttributes, listItemChildrenTabIndex, listItemClassList});
-  }
-
   get classes() {
-    const {className, nonInterative, dense, avatarList, twoLine} = this.props;
+    const {className, nonInteractive, dense, avatarList, twoLine} = this.props;
     return classnames('mdc-list', className, {
-      'mdc-list--non-interactive': nonInterative,
+      'mdc-list--non-interactive': nonInteractive,
       'mdc-list--dense': dense,
       'mdc-list--avatar-list': avatarList,
       'mdc-list--two-line': twoLine,
     });
   }
 
-  getListItemClasses_(id, className) {
-    const {listItemClassList} = this.state;
-    return listItemClassList[id] ?
-      classnames(Array.from(listItemClassList[id]), className) :
-      className;
-  }
-
   get adapter() {
     return {
-      getListItemCount: () => Object.keys(this.listItems_).length,
-      getFocusedElementIndex: () => this.getListItemIndexOfTarget_(document.activeElement),
+      getListItemCount: () => this.listItemCount,
+      getFocusedElementIndex: () => {
+        let count = -1;
+        let index = -1;
+        React.Children.forEach(this.props.children, (child) => {
```

Comment:
this is cool - didn't know you could inspect child.type.name. Aren't you give the index here? Is it because you're accounting for dividers and other elements that aren't ListItems?

Reply:
Also is there a way to break early from this? If you've found the element

Reply:
yep, the index thing is to get around dividers and non-ListItems. and you can't break early in `React.Children.forEach` so I changed it to use `React.Children.toArray` instead. https://stackoverflow.com/questions/51334473/react-how-to-break-out-of-react-children-foreach-loop